backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/services/GeneralViewNodeToolSectionSwitch.java

```diff
@@ -253,13 +253,19 @@ public List<NodeToolSection> caseOccurrenceDefinition(OccurrenceDefinition objec
     public List<NodeToolSection> casePartDefinition(PartDefinition object) {
         var createSection = this.createPartDefinitionElementsToolSection();
         createSection.getNodeTools().addAll(this.createToolsForCompartmentItems(object));
+        // Remove New Action tool, we use a custom New Action tool from ActionFlowCompartmentNodeToolProvider
+        createSection.getNodeTools().removeIf(nodeTool -> Objects.equals(nodeTool.getName(), "New Action"));
+        createSection.getNodeTools().add(new ActionFlowCompartmentNodeToolProvider(SysmlPackage.eINSTANCE.getPartDefinition(), this.descriptionNameGenerator).create(null));
         return List.of(createSection, this.toolDescriptionService.addElementsNodeToolSection(true));
     }
 
     @Override
     public List<NodeToolSection> casePartUsage(PartUsage object) {
         var createSection = this.createPartUsageElementsToolSection();
         createSection.getNodeTools().addAll(this.createToolsForCompartmentItems(object));
+        // Remove New Action tool, we use a custom New Action tool from ActionFlowCompartmentNodeToolProvider
+        // createSection.getNodeTools().removeIf(nodeTool -> Objects.equals(nodeTool.getName(), "New Action"));
```

Comment:
Is it normal to have this line commented?

Reply:
No it was a mistake, I fixed it